main.pas

```diff
@@ -3301,6 +3389,178 @@ procedure TForm1.DrawZBBRELFile(filename: ansistring; px, py, pz: double);
+    isedited := true;
+  end;
+end;
+
+procedure TForm1.Copylastitem1Click(Sender: TObject);
+var
+x, y: integer;
+begin
+  if (form10.ComboBox1.ItemIndex > -1) and (form10.tag = 1) then
+  begin
+    inc(Floor[sfloor].ObjCount);
+    for x := 0 to preseti - 1 do
+      if ObjTemplate[x].name = form10.ComboBox1.Text then
+        break;
+    for y := 0 to sizeof(TObj) - 1 do
+      pansichar(@Floor[sfloor].Obj[Floor[sfloor].ObjCount - 1])[y] := pansichar(@ObjTemplate[x].data)[y];
+
+    if form10.UnicodestringGrid1.Visible then
+      Floor[sfloor].Obj[Floor[sfloor].ObjCount - 1].unknow8 := strtofloat(form10.UnicodestringGrid1.Cells[1, 0]);
+
+    if form10.UnicodeStringGrid2.Visible then
+    begin
+      Floor[sfloor].Obj[Floor[sfloor].ObjCount - 1].unknow8 := strtofloat(form10.UnicodeStringGrid2.Cells[1, 0]);
+      Floor[sfloor].Obj[Floor[sfloor].ObjCount - 1].unknow9 := strtofloat(form10.UnicodeStringGrid2.Cells[1, 1]);
+      Floor[sfloor].Obj[Floor[sfloor].ObjCount - 1].Unknow10 := strtofloat(form10.UnicodeStringGrid2.Cells[1, 2]);
+    end;
+
+    // Set default position based on user's setting
+    Floor[sfloor].Obj[Floor[sfloor].ObjCount - 1].map_section := FPlacementOptions.seDefaultSect.Value;
+    Floor[sfloor].Obj[Floor[sfloor].ObjCount - 1].Pos_X := FPlacementOptions.nbDefaultX.Value;
+    Floor[sfloor].Obj[Floor[sfloor].ObjCount - 1].Pos_Y := FPlacementOptions.nbDefaultZ.Value;
+    Floor[sfloor].Obj[Floor[sfloor].ObjCount - 1].Pos_Z := FPlacementOptions.nbDefaultY.Value;
+    lblStatus.Visible := true;
+    lblModifiers.Visible := true;
+    MoveSel := Floor[sfloor].ObjCount - 1;
+    MoveType := 2;
+    ListBox2.Items.Add('#' + inttostr(MoveSel) + ' - ' + GetObjName(Floor[sfloor].Obj[MoveSel].Skin));
+    if have3d then
+    begin
+      MyObjCount := Floor[sfloor].ObjCount;
+      setlength(MyObj, MyObjCount);
+      MyObj[MoveSel] := nil;
+      Generateobj(Floor[sfloor].Obj[MoveSel], MoveSel);
+
+    end;
+    ctrldw := true;
+    firstdrop := true;
+    DrawMap;
+    isedited := true;
+  end;
+end;
+
+procedure TForm1.Copylastmonster1Click(Sender: TObject);
+var
+  x, y: integer;
+begin
+  if (form9.ComboBox1.ItemIndex > -1) and (form9.tag = 1) then
+  begin
+    inc(Floor[sfloor].MonsterCount);
+    for x := 0 to presetm - 1 do
+      if MonsterTemplate[x].name = form9.ComboBox1.Text then
+        break;
+    for y := 0 to sizeof(TMonster) - 1 do
+      pansichar(@Floor[sfloor].Monster[Floor[sfloor].MonsterCount - 1])[y] := pansichar(@MonsterTemplate[x].data)[y];
+    Floor[sfloor].Monster[Floor[sfloor].MonsterCount - 1].Unknow5 := form9.SpinEdit1.value;
+    Floor[sfloor].Monster[Floor[sfloor].MonsterCount - 1].unknow6 := form9.SpinEdit1.value;
+    Floor[sfloor].Monster[Floor[sfloor].MonsterCount - 1].unknow3 := MapFloorId[Floor[sfloor].floorid];
+
+    // Set default position based on user's setting
+    Floor[sfloor].Monster[Floor[sfloor].MonsterCount - 1].map_section := FPlacementOptions.seDefaultSect.Value;
```

Comment:
this code is repeated,
you could make a function with it and call it from those different functions